File: LeetCode/Easy/2500-delete-greatest-value-in-each-row/2500-delete-greatest-value-in-each-row.cpp
class Solution {
public:
    int deleteGreatestValue(vector<vector<int>>& grid) {
        int ret = 0;
        int n = grid.size();
        
        for (int i = 0; i < n; ++i) sort(grid[i].begin(), grid[i].end());
        
        int col = grid[0].size() - 1;
        
        while (col > -1){
            int temp = grid[0][col];
            
            for (int row = 1; row < n; ++row) temp = max(temp, grid[row][col]);
            
            ret += temp;
            col--;
        }
        
        return ret;
    }
};

File: LeetCode/Easy/2500-delete-greatest-value-in-each-row/2500-delete-greatest-value-in-each-row.py
class Solution:
    def deleteGreatestValue(self, grid: List[List[int]]) -> int:
        ret = 0
        
        for v in grid:
            v.sort()
            
        while len(grid[0]) > 0:
            temp = 0
            
            for v in grid:
                temp = max(temp, v.pop())
            
            ret += temp
        
        return ret